From efede9597de8e1c63cca08d24f1079283429e1a4 Mon Sep 17 00:00:00 2001
From: Gilberto Garcia <gilbertodamian14@gmail.com>
Date: Tue, 23 Apr 2019 21:58:04 -0500
Subject: [PATCH] Fix errors related with guice Well, with the old order(first
 create the bind, later install bindings) you can't really inject something in
 the main class that's binded in the configure(ProtectedBinder) method because
 this injects the plugin and later create the bindings


diff --git a/src/main/java/org/bukkit/plugin/PluginModule.java b/src/main/java/org/bukkit/plugin/PluginModule.java
index 56bbb8e9..07e6ae6e 100644
--- a/src/main/java/org/bukkit/plugin/PluginModule.java
+++ b/src/main/java/org/bukkit/plugin/PluginModule.java
@@ -13,8 +13,8 @@ public class PluginModule extends ProtectedModule {
 
     @Override
     public void configure() {
-        bind(Plugin.class).toInstance(plugin);
         plugin.configure(binder());
+        bind(Plugin.class).toInstance(plugin);
     }
 
     @Provides
-- 
2.21.0.windows.1

